doc/getting_started_rtkernel.rst

```diff
@@ -0,0 +1,184 @@
+root folder for the repo is assumed to be an absolute or relative path in an
+environment variable named repo.
+
+Create a build directory (on same level as p-net directory) and run cmake::
+
+    mkdir build
+    cd build
+    export COMPILERS=/opt/rt-tools/compilers
+    export RTK=<PATH TO YOUR MODIFIED>/rt-kernel-xmc4/
+    export BSP=xmc48relax
+
+The cmake executable is assumed to be in your path. Run::
+
+    cmake ../p-net -DCMAKE_TOOLCHAIN_FILE=../p-net/cmake/toolchain/rt-kernel-xmc4.cmake -DLOG_LEVEL=DEBUG -DBUILD_TESTING=OFF -DCMAKE_ECLIPSE_EXECUTABLE=/opt/rt-tools/workbench/Workbench -DCMAKE_ECLIPSE_GENERATE_SOURCE_PROJECT=TRUE -G "Eclipse CDT4 - Unix Makefiles"
+
+Depending on how you installed cmake, you might need to run ``snap run cmake``
+instead of ``cmake``.
+
+After running ``cmake`` you can run ``ccmake`` or ``cmake-gui`` to change settings.
+
+Start Workbench::
+
+    /opt/rt-tools/workbench/Workbench
+
+Two Workbench projects have been created by cmake. One project is for building
+the code, the other is for the p-net source code. You need at least to
+import the project for building the code.
+
+Use the menu "File > Import". Select "General > Existing Projects". Click
+"Browse" and select the ``build`` directory that was created earlier.
+
+Use the menu "Project > Build All" to build it.
+
+
+More Workbench settings
+-----------------------
+If you intend to edit the p-net source code in the Workbench tool, you also
+need to import the p-net source project. The files ``.project`` and
+``.cproject`` have been created in the p-net repo by cmake.
+
+Use the menu "File > Import". Select "General > Existing Projects". Click
+"Browse" and select the ``p-net`` repo directory.
+
+
+Run on target
+-------------
+First start Segger J-link GDB debug server::
+
+    JLinkGDBServerExe
+
+Select "USB" and target device "XMC4800-2048" in the GUI. Use "Little Endian",
+Target interface "SWD" and speed "Auto Selection".
+
+Adjust debugger settings by right-clicking the Profinet build project,
+and selecting "Debug as > Debug configurations". Under Hardware Debugging,
+select Profinet... You might need to double click "Hardware Debugging" if the
+child node does not appear. On the "Startup" tab enter ``monitor reset 0``
+in the "Run commands". Click Apply and Close.
+
+Typically these values  have been automatically entered:
+
+* Tab "Main" C/C++ application: ``pn_dev.elf``.
+* Tab "Debugger". Select debugger type J-Link. GDB command:
+  ``${COMPILERS}/arm-eabi/bin/arm-eabi-gdb``.
+
+Run the compiled code on target by right-clicking the Profinet build project,
+and selecting "Debug as > Hardware debugging". Select ``pn_dev.elf`` and
+click OK. The download progress pop-up window should appear.
+
+To be able to view register content, use the MMR tab in the debug view. Select
+core "XMC4800".
+
+Open a terminal to view debug output from the target, which will appear as
+``/dev/ttyACM0``. For example picocom can be used (add yourself to the
```

Comment:
It may not be ttyACM0, if you have other serial devices attached.

Reply:
Updated